datapackage_pipelines/utilities/resources.py

```diff
@@ -31,6 +34,22 @@ def get_path(descriptor):
     return None
 
 
+def insert_hash_in_path(descriptor, hash):
+    path = descriptor.get('path')
+    if isinstance(path, str):
+        path_list = path.split('/')
+        hashed = os.path.join(hash, path_list.pop(-1))
+        path_list.append(hashed)
+        descriptor['path'] = '/'.join(path_list)
+    if isinstance(path, list):
+        if len(path) > 0:
+            path = path[0]
```

Comment:
There's some unnecessary code duplication here.
We can have path to be a string always (regardless of its initial form)